src/box/iproto_features.h

```diff
@@ -77,6 +77,12 @@ extern "C" {
 	 * tuple formats are received in IPROTO_TUPLE_FORMATS field.
 	 */								\
 	_(CALL_ARG_TUPLE_EXTENSION, 9)					\
+	/**
+	 * Checkpoint (from files) FETCH_SNAPSHOT request support:
+	 * IPROTO_IS_CHECKPOINT_JOIN, IPROTO_CHECKPOINT_VCLOCK and
+	 * IRPOTO_CHECKPOINT_LSN in FETCH_SNAPSHOT.
+	 */								\
+	_(CHECKPOINT_FETCH_SNAPSHOT, 10)				\
```

Comment:
The name is kinda awkward. I think we'd better name it `FETCH_SNAPSHOT_CURSOR` to emphasize that it's an extension of the `IPROTO_FETCH_SNAPSHOT` request.

Also, I think we should rename `ENABLE_CHECKPOINT_JOIN` to `ENABLE_FETCH_SNAPSHOT_CURSOR` for consistency and because `IPROTO_JOIN` doesn't support the feature (initially, I thought it'd support it). Using `checkpoint_join` in engine callback names is fine because they should work both for `IPROTO_FETCH_SNAPSHOT` and `IPROTO_JOIN`.